src/network/networkd-dhcp-common.c

```diff
@@ -505,12 +544,34 @@ int config_parse_section_route_table(
                 return 0;
         }
 
-        if (STRPTR_IN_SET(section, "DHCP", "DHCPv4")) {
+        switch(ltype) {
+        case AF_INET:
```

Comment:
wouldn't it make sense to to change this to AF_INET+RTPROT_DHCP too here, just to be fully systematic?